cfgmgr/vxlanmgr.cpp

```diff
@@ -409,9 +470,138 @@ bool VxlanMgr::doVxlanTunnelMapCreateTask(const KeyOpFieldsValuesTuple & t)
 
     std::string vxlanTunnelMapName = kfvKey(t);
     std::replace(vxlanTunnelMapName.begin(), vxlanTunnelMapName.end(), config_db_key_delimiter, delimiter);
-    m_appVxlanTunnelMapTable.set(vxlanTunnelMapName, kfvFieldsValues(t));
+
+    if (m_vxlanTunnelMapCache.find(vxlanTunnelMapName) != m_vxlanTunnelMapCache.end())
+    {
+        SWSS_LOG_ERROR("Map already present : %s", vxlanTunnelMapName.c_str());
+        return true;
+    }
 
     SWSS_LOG_NOTICE("Create vxlan tunnel map %s", vxlanTunnelMapName.c_str());
+    std::string vlan, vlan_id, vni_id, src_ip, dst_ip("");
+    for (auto i : kfvFieldsValues(t))
+    {
+        const std::string & field = fvField(i);
+        const std::string & value = fvValue(i);
+        if (field == VLAN)
+        {
+            vlan = value;
+        }
+        else if (field == VNI)
+        {
+            vni_id = value;
+        }
+    }
+
+    // Check for VLAN or VNI if they are already mapped
+    if (m_vlanMapCache.find(vlan) != m_vlanMapCache.end())
+    {
+        SWSS_LOG_ERROR("Vlan %s already mapped. Map Create failed for : %s", 
+                      vlan.c_str(), vxlanTunnelMapName.c_str());
+        return true;
+    }
+
+    if (m_vniMapCache.find(vni_id) != m_vniMapCache.end())
```

Comment:
m_vniMapCache maintains only vlan <--> VNI map cache. Should we check for VRF <--> VNI mapper as well? There is no cache maintained as of now for that mapper in vxlanmgr.

Reply:
VNI-VRF Mapping is being handled in the VrfMgr.